compilers/concrete-compiler/compiler/lib/Dialect/FHE/Analysis/ConcreteOptimizer.cpp

```diff
@@ -279,9 +324,38 @@ struct FunctionToDag {
     // TODO: use APIFloat.sqrt when it's available
     double manp = sqrt(smanp_int.getValue().roundToDouble());
     auto comment = std::string(op.getName().getStringRef()) + " " + loc;
-    index[val] =
-        dagBuilder.add_levelled_op(slice(inputs), lwe_dim_cost_factor,
-                                   fixed_cost, manp, slice(out_shape), comment);
+
+    double maxInputManp = 0.;
+    size_t n_inputs = 0;
+    for (auto input : op.getOperands()) {
+      if (!fhe::utils::isEncryptedValue(input)) {
+        continue;
+      }
+      n_inputs += 1;
+      if (input.isa<mlir::BlockArgument>()) {
+        maxInputManp = fmax(1., maxInputManp);
+      } else {
+        auto inpSmanpInt =
+            input.getDefiningOp()->getAttrOfType<mlir::IntegerAttr>("SMANP");
+        const double inpManp = sqrt(inpSmanpInt.getValue().roundToDouble());
+        maxInputManp = fmax(inpManp, maxInputManp);
+      }
+    }
+    assert(inputs.size() == n_inputs);
```

Comment:
to do before loop
```assert(inputs.size() == op.getNumOperands())```

Reply:
Some operands may not be included in `n_inputs` (if they are not encrypted datatypes for instance.